Implementing AssetBundles Browser Client

heyaudace opened the pull request:
Implemented AssetBundles Browser Client and Changed the "Create Bundle" section in the UI to accomodate this

#### With AssetBundle Browser present:
<img width="684" alt="browser_present" src="https://user-images.githubusercontent.com/18396695/42644069-81a4fcd6-85af-11e8-8e9b-d2cd4743b51d.png">

#### With AssetBundle Browser not present:

<img width="712" alt="browser_absent" src="https://user-images.githubusercontent.com/18396695/42644126-a4909156-85af-11e8-9507-0c5a996c0399.png">

heyaudace committed:
Implemented AssetBundle Browser Client and Changed the UI to make it responsive to the browser

heyaudace committed:
Implemented AssetBundle Browser Client and Changed the UI to make it responsive to the browser

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+                foreach (var type in assembly.GetTypes())
+                {
+                    // Look for AssetBundleBrowserMain in the AssetBundleBrowser Namespace
+                    if (type != null && type.Namespace != null && type.Namespace.Equals("AssetBundleBrowser") &&
```
Replace
type.Namespace != null && type.Namespace.Equals("AssetBundleBrowser")
with
string.Equals(type.Namespace, "AssetBundleBrowser")

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+        // Detects AssetBundleBrowser Namespace and the AssetBundleBrowserMain Class
+        public static bool BundleBrowserIsPresent()
+        {
+            if (_assetBundleBrowserIsPresent.HasValue) return _assetBundleBrowserIsPresent.Value;
```
Go ahead and wrap return in {}

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+            return false;
+        }
+
+        public static string BundleBrowserVersion => GetBrowserVersion(GetAssetBundlesBrowserPackageDotSONFiles()); 
```
Please move to top with other fields.

Can this be readonly?

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+        {
+            var assetsPath = Application.dataPath;
+            var files = Directory.GetFiles(assetsPath, "package.json", SearchOption.AllDirectories);
+            return files.Where(f => f.Contains("AssetBundle") && f.Contains("Browser")).ToArray();
```
1) What is the exact path you're trying to match? Can we be more exact here?

2) Can there be more than 1 match and if so is the version even well defined? Can we just return a single file path?

heyaudace replied:
"Resolved".  Made it so that one looks into package.json in folders with AssetBundles-Browser in their name

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+        
+
+        // Get package.json files related to AssetBundles Browser
+        private static string[] GetAssetBundlesBrowserPackageDotSONFiles()
```
What is SON? JSON? If so, use "Json"

heyaudace replied:
My bad! Typo

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+        // Goes line by line since there is no guarantee that any given file will be a valid well formatted JSON file
+        private static string GetBrowserVersion(string[] filePaths)
+        {
+            foreach (var filePath in filePaths)
```
At a glance it looks like this code might be parsing every file rather than just the package.json file.

Maybe combine the two methods?

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+            foreach (var filePath in filePaths)
+            {
+                var allText = File.ReadAllText(filePath);
+                if (!allText.ToLower().Contains("\"name\": \"com.unity.assetbundlebrowser\"")) continue;
```
What is the string we're trying to get here? Can we use a regex to capture the part you need?

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -68,14 +69,29 @@ void OnGUI()
+                                       "and bundle it (and its dependencies) into an AssetBundle file.", EditorStyles.wordWrappedLabel);
             EditorGUILayout.Space();
-            EditorGUILayout.LabelField(string.Format("AssetBundle Browser version: {0}", "not found"), EditorStyles.wordWrappedLabel);
+            EditorGUILayout.LabelField($"AssetBundle Browser version: { AssetBundleBrowserClient.BundleBrowserVersion }",  EditorStyles.wordWrappedLabel);
```
Does https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/interpolated work with .NET 3.5? Make sure you test with that version instead of 4.X

Name "Asset Bundle Browser"

heyaudace replied:
Resolved. Now working in 3.5 to make sure that everything is compatible

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -68,14 +69,29 @@ void OnGUI()
+                var downloadAssetBundleBrowser =  GUILayout.Button ("Download AssetBundles Browser", GUILayout.Width(ButtonWidth));
+                if (downloadAssetBundleBrowser)
+                {
+                    Process.Start("https://docs.unity3d.com/Manual/AssetBundles-Browser.html");
```
Application.OpenURL() instead

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,101 @@
+        {
+            if (!BundleBrowserIsPresent())
+            {
+                throw new Exception("Cannot detect AssetBundleBrowser Context");
```
How about "Cannot detect Unity Asset Bundle Browser"

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -68,14 +69,29 @@ void OnGUI()
             EditorGUILayout.LabelField("AssetBundle Creation", EditorStyles.boldLabel);
-            EditorGUILayout.LabelField("Use the Unity Asset Bundle Browser to select your game's main scene " + 
-                "and bundle it (and its dependencies) into an AssetBundle file.", EditorStyles.wordWrappedLabel);
+            EditorGUILayout.LabelField("Use the Unity AssetBundles Browser to select your game's main scene " + 
```
Switch back to "Unity Asset Bundle Browser" since that's the name they use

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -68,14 +69,29 @@ void OnGUI()
+                }
+            } else 
+            {
+                var downloadAssetBundleBrowser =  GUILayout.Button ("Download AssetBundles Browser", GUILayout.Width(ButtonWidth));
```
Link to the github releases page directly to save some time?

Let's have 2 buttons maybe:
"Open Asset Bundle Browser Documentation" (the link you have now)
and
"Download Asset Bundle Browser from GitHub" (the github releases page)

heyaudace committed:
Addressing code review

heyaudace commented:
## Addressed all the feedback from the code review

### UI With AssetBundle Present:
<img width="696" alt="screen shot 2018-07-12 at 4 24 34 pm" src="https://user-images.githubusercontent.com/18396695/42664556-50f5e794-85f0-11e8-81fe-9e15f6e43270.png">



### UI With AssetBundle Absent:

<img width="696" alt="screen shot 2018-07-12 at 4 25 41 pm" src="https://user-images.githubusercontent.com/18396695/42664565-59fc4c5c-85f0-11e8-901d-2033b5187d40.png">

#### Changed the syntax to support unity 3.5 as well.

heyaudace committed:
Addressing code review #2

heyaudace committed:
removing unnecessary spaces

leegao commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,116 @@
+                        type.Name.Equals("AssetBundleBrowserMain"))
+                    {
+                        _assetBundleBrowserIsPresent = true;
+                        return _assetBundleBrowserIsPresent.Value;
```
Just a comment: I'm totally surprised these aren't unboxed by default, but I can see why this makes sense; assignments aren't as dangerous as uses of nullables, and this is actually a very forward-looking language design

leegao commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,116 @@
+        private static bool IsAssetBundleBrowserFolder(string folderName)
+        {
+            const string pattern = @"AssetBundles-Browser";
+            var rx = new Regex(pattern, RegexOptions.Compiled | RegexOptions.IgnoreCase);
```
nit: let's either use the name regex here or pattern, let's also either inline the @"AssetBundles-Browser" above into this call or surface it as a class-level constant.

thanwah replied:
I read a bit more about this:
https://docs.microsoft.com/en-us/dotnet/standard/base-types/best-practices#ObjectInstantiation

For your use case it looks like the best option is:
Regex.IsMatch(folderName, "AssetBundles-Browser", RegexOptions.IgnoreCase)

https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.ismatch?view=netframework-4.7.2#System_Text_RegularExpressions_Regex_IsMatch_System_String_System_String_System_Text_RegularExpressions_RegexOptions_

heyaudace replied:
Very Cool.

Allows me to get rid of the entire method itself.  Thanks & Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -49,53 +55,62 @@ public static bool BundleBrowserIsPresent()
-                string line;
-                while ((line = fileStream.ReadLine()) != null)
+                var expectedPackageDotJsonPath = Path.Combine(folderPath, "package.json");
+                if (!File.Exists(expectedPackageDotJsonPath)) continue;
```
Nit: wrap in braces

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -49,53 +55,62 @@ public static bool BundleBrowserIsPresent()
+        // Suppress warnings about non-initialization of fields
+#pragma warning disable CS0649 
+        [Serializable]
+        private class PackageDotJSsonContents
```
Remove extra "S" for PackageDotJsonContents

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -69,26 +70,33 @@ void OnGUI()
                     AssetBundleBrowserClient.DisplayAssetBundleBrowser();
                 }
-            } else 
+            } else
```
The "}" and "else" should be on different lines I think. Use the code reformatter?

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -69,26 +70,33 @@ void OnGUI()
+            } else
             {
-                var downloadAssetBundleBrowser =  GUILayout.Button ("Download AssetBundles Browser", GUILayout.Width(ButtonWidth));
+                var downloadAssetBundleBrowser =  GUILayout.Button ("Download Asset Bundle Browser from GitHub", GUILayout.Width(LongButtonWidth));
```
Seeing an extra " " after "=" and oanother before "(". Please re-run the code formatter.

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -69,26 +70,33 @@ void OnGUI()
+                }
+                EditorGUILayout.Space();
+                
+                var viewAssetBundleBrowserDocumentation =  GUILayout.Button ("Open Asset Bundle Browser Documentation", GUILayout.Width(LongButtonWidth));
```
same

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -69,26 +70,33 @@ void OnGUI()
             if (AssetBundleBrowserClient.BundleBrowserIsPresent())
             {
-                var openAssetBundleBrowser = GUILayout.Button ("Open AssetBundles Browser", GUILayout.Width(ButtonWidth));
+                var openAssetBundleBrowser = GUILayout.Button ("Open Asset Bundle Browser", GUILayout.Width(ButtonWidth));
```
remove space before (

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -49,53 +55,62 @@ public static bool BundleBrowserIsPresent()
+                        return json.version;
                     }
                 }
+                catch (Exception)
```
Let's not catch generic exceptions, especially without any sort of logging.

What is the specific exception you're trying to handle? Is there another way to handle it?

heyaudace replied:
Resolved.

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -49,53 +55,62 @@ public static bool BundleBrowserIsPresent()
             EditorApplication.ExecuteMenuItem(AssetBundleBrowserMenuItem);
         }
+
+        // Allows to create a serializable object from containing only name and version attributes from package.json
```
How about

Represents name and version fields from the package.json file of the Asset Bundle Browser project:
https://github.com/Unity-Technologies/AssetBundles-Browser/blob/master/package.json
Suppress warnings about non-initialization of fields.

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -49,53 +55,62 @@ public static bool BundleBrowserIsPresent()
+        private class PackageDotJSsonContents
+        {
+            public string version;
+            public string name;
```
Nit: flip the order of these 2 fields with name first?

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -15,28 +15,34 @@
-
+        private const string AssetBundleBrowserName = "com.unity.assetbundlebrowser";
+        private const string AssetBundleBrowserMenuItem = "Window/AssetBundle Browser";
+        public static readonly string AssetBundleBrowserVersion = GetBrowserVersion();
```
What if the developer updates the installed version? How will this be updated?

Note: updating every time the editor calls OnGUI() may be too expensive.

heyaudace replied:
Resolved

For when version is updated and reducing the performance cost, I made changes such that computations for detecting asset bundle browser and its version are done only when the "Create Bundle" tab is clicked. This way, the developer can just close and reopen quick deploy window, or simply open another tab and come back to this one for the changes to be reflected.

Got  a question tho! @thanwah I'm Wondering if we could just add  a small refresh button instead of recomputing. This would reduce the cost of computation even more, assuming that the developer will rarely update the installed version.

thanwah reviewed (commented):
Looking very good. A few more comments.

heyaudace committed:
Addressing Review Comments

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -65,24 +72,55 @@ void OnGUI()
+            }
+            else
+            {
+                var downloadAssetBundleBrowser = GUILayout.Button("Download Asset Bundle Browser from GitHub",
```
Nit: inline this

heyaudace replied:
Resolved

thanwah replied:
Not inlined?

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -65,24 +72,55 @@ void OnGUI()
+
+                EditorGUILayout.Space();
+
+                var viewAssetBundleBrowserDocumentation = GUILayout.Button("Open Asset Bundle Browser Documentation",
```
Nit: inline this variable

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -14,15 +14,20 @@
 
 using UnityEditor;
 using UnityEngine;
+using System.Diagnostics;
```
Is this still needed? Run the IDE's Code -> Optimize Imports option on each file you modified.

heyaudace replied:
Resolved.

thanwah commented on GooglePlayInstant/Editor/PlayInstantQuickDeployWindow.cs:
```diff
@@ -65,24 +72,55 @@ void OnGUI()
+            // Allow the developer to open the AssetBundles Browser if it is present, otherwise ask them to download it
+            if (AssetBundleBrowserClient.AssetBundleBrowserIsPresent)
+            {
+                var openAssetBundleBrowser =
```
Nit: inline variable

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+        // Suppress warnings about non-initialization of fields.
+#pragma warning disable CS0649 
+        [Serializable]
+        private class PackageDotJSsonContent
```
JSson -> Json

heyaudace replied:
Resolved.

Thanks. Hard  to see these

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+            _assetBundleBrowserVersion = GetBrowserVersion(false);
+        }
+
+        // Represents name and version fields from the package.json file of the Asset Bundle Browser project:
```
Put class documetnatino in a /// <summary> block

heyaudace replied:
@thanwah  This is a private class so I decided to keep it in comments.

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+
+        // CallBack method to re-update static values when this tab is re-opened
+        /// <summary>
+        /// Reload and update Asset Bundle Browser information
```
This comment doesn't add any information beyond what the method is already named.

Maybe take your floating comment in line 146 and replace this with that.

heyaudace replied:
Resolved.
Added more useful information

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+        {
+            if (!BundleBrowserIsPresent())
+            {
+                Debug.LogError("Cannot detect Unity Asset Bundle Browser");
```
Missing a "return;" ?

heyaudace replied:
Resolved!

Yes! Was initially throwing an error so I changed to a warning and forgot to add a return

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+        }
+
+        /// <summary>
+        /// Display the Asset Bundle Browser Window
```
Display the Asset Bundle Browser window or log an error if not present.

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+                }
+                catch (ArgumentException e)
+                {
+                    Debug.LogWarning(string.Format(
```
Use Debug.LogWarningFormat( instead of Debug.LogWarning(string.Format(

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+        {
+            get { return GetBrowserVersion(); }
+        }
+
```
Nit: delete newline

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+        private static string _assetBundleBrowserVersion;
+
+        /// <summary>
+        /// Whether or not Asset Bundle Browser is present
```
General nit: try to be consistent in documentation, e.g. always end with "." (you don't here but do online 40) and put class/method documentation in summary blocks. Also try to say something more useful than just repeating the name of the method in English.

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+        /// <summary>
+        /// Whether or not Asset Bundle Browser is present
+        /// </summary>
+        public static bool AssetBundleBrowserIsPresent
```
Delete this method and make the one it calls public? It doesn't seem to do anything beyond the one it calls

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+        /// <summary>
+        /// The detected version of Asset Bundle Browser.
+        /// </summary>
+        public static string AssetBundleBrowserVersion
```
Delete this method and make the one it calls public? It doesn't seem to do anything beyond the one it calls

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+using System.Linq;
+using System.Text.RegularExpressions;
+using UnityEditor;
+using UnityEngine;
```
Use Code -> Optimize Imports on each file

heyaudace replied:
Resolved

heyaudace commented:
Addressed code comments. No Changes on the UI in the most recent commit

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+
+namespace GooglePlayInstant.Editor
+{
+    public static class AssetBundleBrowserClient
```
Add class "summary" documentation.

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+{
+    public static class AssetBundleBrowserClient
+    {
+        private const string AssetBundleBrowserName = "com.unity.assetbundlebrowser";
```
AssetBundleBrowserPackageName

heyaudace replied:
Resolved

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                // Checking that assembly != null since we do not want to rely on iteration through
+                // AppDomain.CurrentDomain.GetAssemblies() to yield non-null items for this function to work
```
Has this happened to you or do you have documentation saying it can happen?

heyaudace replied:
Resolved.

Was trying to be a little more defensive here but doesn't make sense to have a null assembly. Removed the null check

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+                foreach (var type in assembly.GetTypes())
+                {
+                    // Look for AssetBundleBrowserMain in the AssetBundleBrowser Namespace
+                    if (type != null && string.Equals(type.Namespace, "AssetBundleBrowser") &&
```
Can type be null?

heyaudace replied:
Resolved.

Was trying to be a little more defensive here as well, but again I don't think it makes sense. Removed

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+            {
+                var expectedPackageDotJsonPath = Path.Combine(folderPath, "package.json");
+                if (!File.Exists(expectedPackageDotJsonPath))
+                {
```
Log warning?

heyaudace replied:
Resolved.
Added a Debug.LogWarning just so the the developer knows we attempted to find the file

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,167 @@
+                try
+                {
+                    var json = JsonUtility.FromJson<PackageDotJSsonContent>(data);
+
```
Nit: delete newline

heyaudace replied:
Resolved

heyaudace committed:
Addressing Code Review comments

heyaudace reviewed (commented):
Addressed all the review comments.

heyaudace committed:
Addressed the Regex comment

heyaudace commented:
All comments now resolved, and no changes to the UI

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,151 @@
+            // All forlders with "AssetBundles-Browser" in their names are considered candidates
+            var assetBundleBrowserFolderPaths =
+                Directory.GetDirectories(Application.dataPath).ToArray().Where(folderName =>
+                    Regex.IsMatch(folderName, "AssetBundles-Browser", RegexOptions.IgnoreCase));
```
Nice to see this inlined!

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,151 @@
+            }
+
+            // Extract AssetBundleBrowser version name from the Asset Bundle Browser package.json file.
+            // All forlders with "AssetBundles-Browser" in their names are considered candidates
```
Nit: forlders -> folders

thanwah commented on GooglePlayInstant/Editor/PlayInstantAssetBundleBrowserClient.cs:
```diff
@@ -0,0 +1,151 @@
+        private static bool? _assetBundleBrowserIsPresent;
+        private static string _assetBundleBrowserVersion;
+
+        /// <param name="useCurrentValueIfPresent">A boolean value corresponding to whether the caller wants to
```
Wrap the whole thing in a summary block. I'm not sure the param makes sense without that.

thanwah reviewed (approved):
Just a few minor comments to address. Otherwise LGTM. Thanks!

heyaudace committed:
Addressing final comments

heyaudace committed:
Merge branch 'master' into assetbundles_browser
